kernel/kernel-api/src/main/java/io/delta/kernel/internal/snapshot/SnapshotManager.java

```diff
@@ -224,6 +225,45 @@ public void checkpoint(Engine engine, long version) throws TableNotFoundExceptio
     logger.info("{}: Last checkpoint metadata file is written for version: {}", tablePath, version);
 
     logger.info("{}: Finished checkpoint for version: {}", tablePath, version);
+    doLogCleanup(engine, checkpointMetaData, snapshot);
+  }
+
+  private void doLogCleanup(
+      Engine engine, CheckpointMetaData checkpointMetaData, SnapshotImpl snapshot)
+      throws IOException {
+    Metadata metadata = snapshot.getMetadata();
+    Boolean enableExpireLogCleanup =
+        TableConfig.ENABLE_EXPIRED_LOG_CLEANUP.fromMetadata(engine, metadata);
+    if (enableExpireLogCleanup) {
+      Long retentionMillis = TableConfig.LOG_RETENTION.fromMetadata(engine, metadata);
+      Long fileCutOffTime = System.currentTimeMillis() - retentionMillis;
+      logger.info(
+          "{}: Starting the deletion of log files older than {}", tablePath, fileCutOffTime);
+      int numDeleted = 0;
+      try (CloseableIterator<FileStatus> files =
```

Comment:
I think we should check there atleast one checkpoint to after the latest deleted file to make sure the table is still time travelable.

```
008.json
009.json
010.json
010.checkpoint.parquet
011.json
012.json
013.json
014.json
...
020.json
020.checkpoint.parquet
```

If the retention selection 8 to 13.json files, we should only delete up to the 10.json.

Reply:
this is fixed in the latest commit